pilot/pkg/networking/grpcgen/lds.go

```diff
@@ -122,6 +123,12 @@ func buildInboundListeners(node *model.Proxy, push *model.PushContext, names []s
 func buildInboundFilterChains(node *model.Proxy, push *model.PushContext, si *model.ServiceInstance, applier authn.PolicyApplier) []*listener.FilterChain {
 	mode := applier.GetMutualTLSModeForPort(si.Endpoint.EndpointPort)
 
+	// auto-mtls label is set - clients will attempt to connect using mtls, and
+	// gRPC doesn't support permissive.
+	if node.Metadata.Labels[label.SecurityTlsMode.Name] == "istio" && mode == model.MTLSPermissive {
+		mode = model.MTLSStrict
```

Comment:
Doesn't this break clients that have no sidecar?

Reply:
Why would it break them ? If a client without sidecar doesn't have the auto-mtls label, grpc will be configured as plain text - and  clients will use plain text as well ( as is the default behavior with auto-mtls ).

If the client has the auto-mtls label - we configure TLS, and clients will work as expected. 

It is still possible to set DestinationRule and PeerAuth explicitly to either STRICT or DISABLED - however 
permissive is not supported by gRPC, but we can do auto mtls.

Reply:
If you mean regular gRPC that is not using XDS ( i.e. no sidecar, no proxyless gRPC ) - yes, they would be broken if 
auto-mtls label is set. 

Since proxyless doesn't support permissive - in order to migrate the user will need to use a separate port, and make code  changes if they want to transition. The purpose of this PR is to make sure Istio workloads work properly with automtls, having non-mesh gRPC clients talk with mesh enabled is not supported unless 2 ports are used.

Note that when we implement 'BTS/Hbone', we will have a dedicated port for mTLS and the original port will be plain text. This is not implemented, so we still need auto-mtls to work properly.

Reply:
>If a client without sidecar doesn't have the auto-mtls label, grpc will be configured as plain text - and clients will use plain text as well ( as is the default behavior with auto-mtls ).

I understand `node.Metadata.Labels[label.SecurityTlsMode.Name] == "istio"`  is saying the grpc server is injected with a sidecar.

Reply:
If grpc has sidecar though not common, then should always disable tls?

Reply:
I think the label is saying 'workload supports istio TLS mode' - not that it has a sidecar ( or an envoy sidecar ). 
The mode really means mTLS connections with spifee certs, etc.

gRPC supports the same auth as Istio - same certs, verification, etc. We certainly don't want to disable it. 

It is possible for a proxyless gRPC application to have a sidecar - with the gRPC port excluded from interception. 
In this case - the label will be set, and gRPC ( like the intercepted ports ) will use the exact same mTLS.

Reply:
I think i get to understand the case, grpc does not support permissive mode, so need to do a special processing